sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RequestTimeline.java

```diff
@@ -172,11 +169,12 @@ public Event(final String name, final Instant from, final Instant to) {
             this.name = name;
             this.startTime = from;
 
-            this.duration = from == null ? null : to == null ? Duration.ZERO : Duration.between(from, to);
-            if(this.duration != null) {
-                this.durationInMicroSec = duration.toNanos()/1000L;
+            if (from == null) {
+                this.duration = null;
+            } else if (to == null) {
+                this.duration = Duration.ZERO;
```

Comment:
Right interpretation of NULL and ZERO is hard. 
How about populate Duration only when end-time is present?

Or even better is list Start and EndTimes

Reply:
I'll consider this change after this PR.